Sources/Sentry/SentrySdkInfo.m

```diff
@@ -72,43 +56,41 @@ - (instancetype)initWithDictInternal:(NSDictionary *)dict orDefaults:(SentrySdkI
         } else if (info && info.version) {
             version = info.version;
         }
-    }
 
-    return [self initWithName:name andVersion:version];
-}
-
-- (nullable NSString *)getPackageName:(SentryPackageManagerOption)packageManager
-{
-    switch (packageManager) {
-    case SentrySwiftPackageManager:
-        return @"spm:getsentry/%@";
-    case SentryCocoaPods:
-        return @"cocoapods:getsentry/%@";
-    case SentryCarthage:
-        return @"carthage:getsentry/%@";
-    default:
-        return nil;
+        if ([sdkInfoDict[@"packages"] isKindOfClass:[NSArray class]]) {
+            NSMutableSet *newPackages = [NSMutableSet set];
+            for (id maybePackageDict in sdkInfoDict[@"packages"]) {
+                if ([maybePackageDict isKindOfClass:[NSDictionary class]]) {
+                    SentrySdkPackage *package =
+                        [[SentrySdkPackage alloc] initWithDict:maybePackageDict];
+                    if (package != nil) {
+                        [newPackages addObject:package];
+                    }
+                }
+            }
+            packages = newPackages;
+        } else if (info && info.packages) {
+            packages = info.packages;
+        }
     }
+
+    return [self initWithName:name andVersion:version andPackages:packages];
 }
 
 - (NSDictionary<NSString *, id> *)serialize
 {
-    NSMutableDictionary *sdk = @{
-        @"name" : self.name,
-        @"version" : self.version,
-    }
-                                   .mutableCopy;
-    if (self.packageManager != SentryPackageManagerUnkown) {
-        NSString *format = [self getPackageName:self.packageManager];
-        if (format != nil) {
-            sdk[@"packages"] = @{
-                @"name" : [NSString stringWithFormat:format, self.name],
-                @"version" : self.version
-            };
```

Comment:
Based on the development docs this should be an array, I don't know if this value was used anywhere, so I just updated it to the array.

If it's not used I would consider removing it completely and only create packages in SentryClient (same as features, integrations...).